securedrop_client/sync.py

```diff
@@ -0,0 +1,76 @@
+import logging
+
+from PyQt5.QtCore import pyqtSignal, QObject, QThread, QTimer, Qt
+from sqlalchemy.orm import scoped_session
+from sdclientapi import API, RequestTimeoutError
+
+from securedrop_client.api_jobs.sync import MetadataSyncJob
+from securedrop_client.crypto import GpgHelper
+
+
+logger = logging.getLogger(__name__)
+
+
+class ApiSync(QObject):
+    '''
+    ApiSync continuously executes a MetadataSyncJob, waiting 15 seconds between jobs.
+    '''
+    sync_started = pyqtSignal()
+    sync_success = pyqtSignal()
+    sync_failure = pyqtSignal(Exception)
+
+    TIME_BETWEEN_SYNCS_MS = 1000 * 15  # fifteen seconds between syncs
+
+    def __init__(
+        self, api_client: API, session_maker: scoped_session, gpg: GpgHelper, data_dir: str
+    ):
+        super().__init__()
+
+        self.api_client = api_client
+        self.session_maker = session_maker
+        self.gpg = gpg
+        self.data_dir = data_dir
+
+        self.sync_thread = QThread()
+        self.sync_thread.started.connect(self._sync)
```

Comment:
heads up the syncs here are running in the main thread: we'd need to ensure that we move the `ApiSync` object to the `sync_thread` (which we can keep a reference to on the controller as we do with the `export_thread`)

Reply:
ah, nice catch!

Reply:
i'll bring it up tomorrow, but the idea was to move all the thread management for sync into ApiSync, the way we do with the ApiJobQueue. the only code that needs to run in a background thread is `_sync` so maybe i could make it a function instead of a method.

Reply:
I came up with a solution be creating a background task class so that we could leave the Controller as is